cross/gdb-7.12/Makefile

```diff
@@ -8,46 +8,40 @@ PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
 DEPENDS = cross/zlib cross/libexpat cross/gmp cross/ncursesw cross/mpfr
 
 HOMEPAGE = https://www.gnu.org/software/gdb/
-COMMENT  = The GNU Project Debugger
+COMMENT  = The GNU Project Debugger.
 LICENSE  = GPLv2/LGPLv2
 
-UNSUPPORTED_ARCHS = $(PPC_ARCHS)
-
 GNU_CONFIGURE = 1
 
 CONFIGURE_TARGET = gdb_configure
 COMPILE_TARGET = gdb_compile
 INSTALL_TARGET = gdb_install
 
-include ../../mk/spksrc.common.mk
-
-CONFIGURE_ARGS += $(TC_CONFIGURE_ARGS)
-CONFIGURE_ARGS += --prefix=$(INSTALL_PREFIX)
-CONFIGURE_ARGS += --enable-host-shared
+CONFIGURE_ARGS  = --enable-host-shared
 CONFIGURE_ARGS += --with-system-zlib
-ifeq ($(call version_ge, ${TCVERSION}, 7.0),1)
-CONFIGURE_ARGS += --localstatedir=$(INSTALL_PREFIX_VAR)
-endif
 
 include ../../mk/spksrc.cross-cc.mk
 
-# Library is not created under armv5 and armv7
+# Library is not created for armv5 and armv7
 ifeq ($(findstring $(ARCH),$(ARMv5_ARCHS) $(ARMv7_ARCHS) $(ARMv7L_ARCHS)),$(ARCH))
 PLIST_TRANSFORM = sed -e 's?lib:lib/libinproctrace.so??g'
-# binary only created on armv7
-else
+endif
+
+ifneq ($(findstring $(ARCH),$(ARM_ARCHS)),$(ARCH))
+# binary only created on arm
 PLIST_TRANSFORM = sed -e 's?bin:bin/run??g'
 endif
 
+
 .PHONY: gdb_configure
 gdb_configure:
 	@$(RUN) mkdir -p build
-	@$(RUN) ; cd build ; PATH=$(TC_PATH):$$PATH CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" CPPFLAGS="$(CPPFLAGS)" CXXFLAGS="$(CXXFLAGS)" $(WORK_DIR)/$(PKG_DIR)/configure $(CONFIGURE_ARGS)
+	@$(RUN) ; cd build ; PATH=$(abspath $(TC_PATH)):$$PATH CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" CPPFLAGS="$(CPPFLAGS)" CXXFLAGS="$(CXXFLAGS)" $(WORK_DIR)/$(PKG_DIR)/configure $(REAL_CONFIGURE_ARGS)
 
 .PHONY: gdb_compile
 gdb_compile:
-	@$(RUN) ; cd build ; PATH=$(TC_PATH):$$PATH make
+	@$(RUN) ; cd build ; PATH=$(abspath $(TC_PATH)):$$PATH make
 
 .PHONY: gdb_install
 gdb_install:
-	@$(RUN) ; cd build ; PATH=$(TC_PATH):$$PATH make install DESTDIR=$(INSTALL_DIR)
+	@$(RUN) ; cd build ; PATH=$(abspath $(TC_PATH)):$$PATH make install DESTDIR=$(INSTALL_DIR)
```

Comment:
I suggest using `realpath` as otherwise `abspath` will provide an empty string if not existant.

Reply:
No, it is the other way round.
abspath does not resolve symlinks and does not require the file names to refer to an existing file or directory.

**$(realpath names…)**

For each file name in names return the canonical absolute name. A canonical name does not contain any . or .. components, nor any repeated path separators (/) or symlinks. In case of a failure the empty string is returned. Consult the realpath(3) documentation for a list of possible failure causes.

**$(abspath names…)**

For each file name in names return an absolute name that does not contain any . or .. components, nor any repeated path separators (/). Note that, in contrast to realpath function, abspath does not resolve symlinks and does not require the file names to refer to an existing file or directory. Use the wildcard function to test for existence. 

taken from https://www.gnu.org/software/make/manual/html_node/File-Name-Functions.html
(or see https://www.cmcrossroads.com/article/gnu-make-path-handling)

Reply:
you are totally right...